pkg/plugin/plugin.go

```diff
@@ -0,0 +1,253 @@
+	if pathExt == "" {
+		pathExt = ".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL"
+	}
+
+	extensions := strings.Split(strings.ToUpper(pathExt), ";")
+	extUpper := strings.ToUpper(ext)
+
+	for _, validExt := range extensions {
+		if validExt == extUpper {
+			return true
+		}
+	}
+	return false
+}
+
+func scanDirectory(dir string) []Plugin {
+	var plugins []Plugin
+
+	entries, err := os.ReadDir(dir)
+	if err != nil {
+		logrus.Debugf("Plugin discovery: failed to scan directory %s: %v", dir, err)
+
+		return plugins
+	}
+
+	for _, entry := range entries {
+		if entry.IsDir() {
+			continue
+		}
+
+		name := entry.Name()
+		if !strings.HasPrefix(name, "limactl-") {
+			continue
+		}
+
+		pluginName := strings.TrimPrefix(name, "limactl-")
+
+		if strings.Contains(pluginName, ".") {
+			ext := filepath.Ext(name)
+			if !isWindowsExecutableExt(ext) {
+				continue
+			}
+
+			pluginName = strings.TrimSuffix(pluginName, ext)
+		}
+
+		fullPath := filepath.Join(dir, name)
+
+		if !isExecutable(fullPath) {
+			continue
+		}
+
+		plugin := Plugin{
+			Name: pluginName,
+			Path: fullPath,
+		}
+
+		if desc := extractDescFromScript(fullPath); desc != "" {
+			plugin.Description = desc
+		}
+
+		plugins = append(plugins, plugin)
+	}
+
+	return plugins
+}
+
+func isExecutable(path string) bool {
+	info, err := os.Stat(path)
+	if err != nil {
+		return false
+	}
+
+	mode := info.Mode()
+	if mode&0o111 != 0 {
+		return true
+	}
+
+	if runtime.GOOS == "windows" {
+		ext := filepath.Ext(path)
+		return isWindowsExecutableExt(ext)
+	}
+
+	return false
+}
+
+func RunExternalPlugin(ctx context.Context, name string, args []string) {
+	if ctx == nil {
+		ctx = context.Background()
+	}
+
+	if err := UpdatePathForPlugins(); err != nil {
+		logrus.Warnf("failed to update PATH environment: %v", err)
+		// PATH update failure shouldn't prevent plugin execution
+	}
+
+	plugins, err := DiscoverPlugins()
+	if err != nil {
+		logrus.Warnf("failed to discover plugins: %v", err)
+		return
+	}
+
+	var execPath string
+	for _, plugin := range plugins {
+		if plugin.Name == name {
+			execPath = plugin.Path
+			break
+		}
+	}
+
+	if execPath == "" {
+		return
+	}
+
+	cmd := exec.CommandContext(ctx, execPath, args...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Env = os.Environ()
+
+	err = cmd.Run()
+	usrlocalsharelima.HandleExitError(err)
+	if err == nil {
+		os.Exit(0) //nolint:revive // it's intentional to call os.Exit in this function
+	}
+	logrus.Fatalf("external command %q failed: %v", execPath, err)
+}
+
+var descRegex = regexp.MustCompile(`<limactl-desc>(.*?)</limactl-desc>`)
+
+func extractDescFromScript(path string) string {
+	content, err := os.ReadFile(path)
+	if err != nil {
+		logrus.Debugf("Failed to read plugin script %s: %v", path, err)
+		return ""
+	}
+
+	if !strings.HasPrefix(string(content), "#!") {
+		logrus.Debugf("Plugin %s: not a script file, skipping description extraction", filepath.Base(path))
```

Comment:
I think the debug log should always include the full path name. This is for debugging issues, normal users will not see the output because they don't use `--debug`.

```suggestion
		logrus.Debugf("Plugin %s: not a script file, skipping description extraction", path)
```

Same for the next error below.